TeleportMenu.lua

```diff
@@ -458,53 +459,36 @@ function tpm:CreateFlyout(flyoutId, iconId, yOffset, name)
 		return flyOutButton
 	end
 
-	-- Loop through spells, either forwards or backwards based on the setting
-	if reverseMageFlyouts and (flyoutId == 1 or flyoutId == 8 or flyoutId == 11 or flyoutId == 12) then
-		for i = spells, 1, -1 do
-			local flyname = nil
-			local spellID = select(1, GetFlyoutSlotInfo(flyoutId, i))
-			if IsSpellKnown(spellID) then
-				for _, v in pairs(dungeons) do
-					if v.id == spellID then
-						flyname = v.name
-					end
-				end
-				flyoutsCreated = flyoutsCreated + 1
-				local flyOutButton = createFlyOutButton(spellID, flyoutsCreated)
-				if db.buttonText == true and flyname then
-					flyOutButton.text = flyOutButton:CreateFontString(nil, "OVERLAY")
-					flyOutButton.text:SetFont("Fonts\\FRIZQT__.TTF", 13, "OUTLINE")
-					flyOutButton.text:SetPoint("BOTTOM", flyOutButton, "BOTTOM", 0, 5)
-					flyOutButton.text:SetText(flyname)
-					flyOutButton.text:SetTextColor(1, 1, 1, 1)
-				end
-				table.insert(flyOutButtons, flyOutButton)
-			end
+	local totalKnownSpells = 0
+	for i = 1, spells do
+		local spellID = select(1, GetFlyoutSlotInfo(flyoutData.id, i))
+		if IsSpellKnown(spellID) then
+			totalKnownSpells = totalKnownSpells + 1
```

Comment:
I see no harm in figuring out how many spells are known.
Perhaps instead it would make sense to create a table variable to avoid asking again and can simply use the table length and the table could contain spellId and short name :thinking: